cgi-bin/LJ/Sysban.pm

```diff
@@ -209,6 +209,12 @@ sub sysban_check {
         return 0 unless scalar @domains >= 2;
         my $domain = "$domains[-2].$domains[-1]";
         return 1 if $check->('email_domain', $domain);
+        # also check for three-element domain bans
```

Comment:
I don't think this will work for cases where you have a 3-element domain in the list, and you want to match a subdomain of it. "The last two parts" or "the last three parts" are both special cases of "longest matching suffix" in my opinion.